x-pack/plugins/snapshot_restore/server/index.ts

```diff
@@ -12,6 +12,7 @@ import { configSchema, SnapshotRestoreConfig } from './config';
 export const plugin = (ctx: PluginInitializerContext) => new SnapshotRestoreServerPlugin(ctx);
 
 export const config: PluginConfigDescriptor<SnapshotRestoreConfig> = {
+  deprecations: ({ deprecate }) => [deprecate('enabled', '8.0.0')],
```

Comment:
We'll need to add a `snapshotRestore.ui.enabled` settings so we can advise users to use that setting instead.